src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.Aot.cs

```diff
@@ -873,10 +875,51 @@ protected override void AllocAndTransformManagedToNative(ILCodeStream codeStream
 
             MethodDesc helper = Context.GetHelperEntryPoint("InteropHelpers", "ConvertManagedComInterfaceToNative");
             LoadManagedValue(codeStream);
-            codeStream.Emit(ILOpcode.ldtoken, emitter.NewToken(this.ManagedParameterType));
-            MethodDesc getTypeFromHandleMethod =
-                Context.SystemModule.GetKnownType("System", "Type").GetKnownMethod("GetTypeFromHandle", null);
-            codeStream.Emit(ILOpcode.call, emitter.NewToken(getTypeFromHandleMethod));
+            CustomAttributeValue<TypeDesc>? guidAttributeValue = (this.ManagedParameterType as EcmaType)?
+                .GetDecodedCustomAttribute("System.Runtime.InteropServices", "GuidAttribute");
+            if (guidAttributeValue == null)
+            {
+                throw new NotSupportedException();
+            }
+
+            var guidValue = (string)guidAttributeValue.Value.FixedArguments[0].Value;
+            var parts = guidValue.Split('-');
+            codeStream.Emit(ILOpcode.ldtoken, emitter.NewToken((EcmaType)this.ManagedParameterType));
+            MetadataType guidType = Context.SystemModule.GetKnownType("System", "Guid");
+            var int32Type = Context.GetWellKnownType(WellKnownType.Int32);
+            var int16Type = Context.GetWellKnownType(WellKnownType.Int16);
+            var byteType = Context.GetWellKnownType(WellKnownType.Byte);
+            guidType.GetParameterlessConstructor();
+            int param1 = int.Parse(parts[0], System.Globalization.NumberStyles.HexNumber);
```

Comment:
It would be simpler and faster to do:

```
byte[] bytes = Guid.Parse(guidValue);
codeStream.EmitLdc(BinaryPrimitives.ReadInt32LittleEndian(bytes));
codeStream.EmitLdc(BinaryPrimitives.ReadInt16LittleEndian(bytes.Slice(4)));
codeStream.EmitLdc(BinaryPrimitives.ReadInt16LittleEndian(bytes.Slice(6)));
for (int i = 8; i < 16; i++)
    codeStream.EmitLdc(bytes[i]);
```